internal/metrics/batching/metrics.go

```diff
@@ -0,0 +1,93 @@
+package batching
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+	api "go.opentelemetry.io/otel/metric"
+
+	"github.com/Azure/tattler/data"
+)
+
+const (
+	subsystem       = "tattler"
+	sourceTypeLabel = "source_type"
+	successLabel    = "success"
+)
+
+var (
+	batchingCount          metric.Int64Counter
+	batchesEmittedCount    metric.Int64Counter
+	batchItemsEmittedCount metric.Int64Counter
+	batchAgeMilliseconds   metric.Int64Histogram
+)
+
+func metricName(name string) string {
+	return fmt.Sprintf("%s_%s", subsystem, name)
+}
+
+// Init initializes the batching metrics.
+func Init(meter api.Meter) error {
+	var err error
+	batchingCount, err = meter.Int64Counter(metricName("batching_total"), api.WithDescription("total number of times tattler handles batching input"))
+	if err != nil {
+		return err
+	}
+	batchesEmittedCount, err = meter.Int64Counter(metricName("batches_emitted_total"), api.WithDescription("total number of batches emitted by tattler"))
+	if err != nil {
+		return err
+	}
+	batchItemsEmittedCount, err = meter.Int64Counter(metricName("batch_items_emitted_total"), api.WithDescription("total number of batch items emitted by tattler"))
+	if err != nil {
+		return err
+	}
+	batchAgeMilliseconds, err = meter.Int64Histogram(
+		metricName("batch_age_ms"),
+		api.WithDescription("age of batch when emitted"),
+		api.WithExplicitBucketBoundaries(50, 100, 200, 400, 600, 800, 1000, 1250, 1500, 2000, 3000, 4000, 5000, 10000),
+	)
+
+	return nil
+}
+
+// RecordBatchingSuccess records successful handling of batch input
+// so we can calculate error rate.
+func RecordBatchingSuccess(ctx context.Context) {
```

Comment:
I should have mentioned this earlier, my mistake.  Here and below we have some really long names.  

I think we can simply go with Sucesss(), Error(), Emitted().  Every import renames this to metrics, so metrics.Success(), metrics.Error(), metrics.Emitted() seems to be more concise and easily readible.  Batching is redundant because your either using "batching.RecordBatchingSuccess" or "metics.RecordBatchingSucces", but it will already be apparent from the scope of the call what success.

This goes for other metric places as well.